lib/honeybadger/plugins/active_job.rb

```diff
@@ -2,7 +2,9 @@ module Honeybadger
   module Plugins
     module ActiveJob
       # Ignore inline and test adapters, as well as the adapters that we support with their own plugins
-      EXCLUDED_ADAPTERS = %i[inline test delayed_job faktory karafka resque shoryuken sidekiq sucker_punch].freeze
+      EXCLUDED_ADAPTERS = %i[inline test delayed_job faktory karafka resque shoryuken sidekiq sucker_punch]
+      # Don't report errors if GoodJob is reporting them
+      EXCLUDED_ADAPTERS << :good_job if defined?(::GoodJob) && !::GoodJob.on_thread_error.nil?
```

Comment:
I'm not 100% on this, but I think you might want to call this in a `requirement` block in the plugin, which is executed after Rails initializers. Although the plugins themselves may also not be required until after initialize (although they technically could be). 

https://github.com/honeybadger-io/honeybadger-ruby/blob/d705037609175555091e4d5ea912aebdb1d3cbd3/lib/honeybadger/init/rails.rb#L32-L34

https://github.com/honeybadger-io/honeybadger-ruby/blob/d705037609175555091e4d5ea912aebdb1d3cbd3/lib/honeybadger/singleton.rb#L62-L67